File: backend/backend/app/schemas/dataset.py
"""Dataset schemas — frozen by contract_reference.json (DatasetOut)."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel


class DatasetOut(BaseModel):
    id: uuid.UUID
    sector: str
    original_filename: str
    status: str
    health_score: float | None = None
    size_bytes: int | None = None
    content_type: str | None = None
    uploaded_by: uuid.UUID
    created_at: datetime
    total_rows: int | None = None
    description: str | None = None

File: frontend/src/types/dataset.ts
// DatasetOut — matches GET /api/v1/datasets in openapi.json.
export interface DatasetMetadata {
  id: string;
  sector: string;
  original_filename: string;
  status: string;
  health_score?: number | null;
  size_bytes?: number | null;
  content_type?: string | null;
  uploaded_by: string;
  created_at: string;
}
